File: test/data/cs/test03/StateInitial.cs
public class StateInitial : BaseState
{
    private StateController _stateController;
    private IControllee _controllee;
    public StateInitial(StateController stateController, IControllee controllee) : base(controllee)
    {
        _stateController = stateController;
        _controllee = controllee;
    }
    public override BaseState? TransitGoInTo()
    {
        _controllee.DoAction0();
        _stateController.InstanceOfNotShooting.Setup(false, false);
        return _stateController.InstanceOfNotShooting;
    }
    public override string GetStateName()
    {
        return "Initial";
    }
    public override BaseState? GetParent()
    {
        return null;
    }
}

File: test/data/typescript/test03/StateInitial.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
export class StateInitial extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public TransitGoInTo(): BaseState | null
    {
        this._controlleeImp.DoAction0();
        this._stateController.InstanceOfNotShooting.Setup(false, false);
        return this._stateController.InstanceOfNotShooting;
    }
    public  GetStateName(): string
    {
        return "Initial";
    }
    public GetParent(): BaseState | null
    {
        return null;
    }
    public GetStateID(): number
    {
        return 3;
    }
}